tests/lib/cli-engine.js

```diff
@@ -643,6 +643,19 @@ describe("CLIEngine", () => {
             assert.equal(report.results[0].messages.length, 0);
         });
 
+        it("should report zero messages when given a config file and a valid file and esprima required as parser", () => {
+
+            engine = new CLIEngine({
+                parser: require("esprima"),
+                useEslintrc: false
+            });
+
+            const report = engine.executeOnFiles(["lib/cli.js"]);
+
+            assert.equal(report.results.length, 1);
+            assert.equal(report.results[0].messages.length, 0);
+        });
+
```

Comment:
Can you also add a test for this in `tests/lib/linter.js`? (Since this also affects the public API when using something like `linter.verify("foo", { parser: { /* some object */ } })`, it would be good to have a test to make sure that works correctly too.